.cloudbuild/scripts/internal/git/deploykey.go

```diff
@@ -0,0 +1,39 @@
+package git
+
+import (
+	"io/ioutil"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+func writeKey(deployKey []byte) (string, error) {
+	keyFile, err := ioutil.TempFile("", "*")
+	if err != nil {
+		return "", trace.Wrap(err, "failed creating keyfile")
+	}
+	defer keyFile.Close()
+
+	// We don't want *anyone else* reading this key
+	err = keyFile.Chmod(0600)
+	if err != nil {
+		return "", trace.Wrap(err, "failed securing deploy key")
+	}
+
```

Comment:
```suggestion
```

It's not relevant here as we're the only thing running at this point, but this pattern in general isn't actually secure, as permissions are checked during `open(2)` rather than at the moment of reading, so if someone opens the file between creation and chmod, they can keep reading it after you write the contents to it. In this case, `ioutil.TempFile` already opens the file with `O_CREAT | O_EXCL` and a mode of `0600`, so we don't need to do anything.

Reply:
Removed